packages/babel-node/src/_babel-node.ts

```diff
@@ -80,12 +99,21 @@ const _eval = function (code: string, filename: string) {
   code = code.trim();
   if (!code) return undefined;
 
+  hasTopLevelAwait = false;
+
   code = babel.transformSync(code, {
     filename: filename,
     ...babelOptions,
+    parserOpts: {
+      allowAwaitOutsideFunction: true,
+    },
     plugins: (opts.plugins || []).concat([replPlugin]),
   }).code;
 
+  if (hasTopLevelAwait) {
```

Comment:
Given that Node.js supports top-level await, could we just leave the await as-is without having to wrap it in an async function?

Reply:
https://github.com/nodejs/node/blob/22ac5976a837e4caab7c3c8f92b211fe2ba29076/lib/internal/repl/await.js#L155-L156

We use `vm` to execute code, which does not support top-level await.
Nodejs also implements it this way, I refer to it. :)